Core/Core/Profile/Settings/PairWithObserverViewController.swift

```diff
@@ -53,17 +68,86 @@ class PairWithObserverViewController: UIViewController, ErrorViewController {
     func generatePairingCode() {
         didGenerateCode = true
         env.api.makeRequest(PostObserverPairingCodes()) { [weak self] response, _, error in
+            performUIUpdate {
+                if let error = error {
+                    self?.spinner.isHidden = true
+                    self?.showError(error)
+                } else {
+                    if ExperimentalFeature.studentQRCodePairing.isEnabled {
+                        self?.generateQRCode(pairingCode: response?.code)
+                    } else {
+                        self?.spinner.isHidden = true
+                        self?.displayPairingCode(response?.code)
+                    }
+                }
+            }
+        }
+    }
+
+    func generateQRCode(pairingCode: String?) {
+        let termsAndConditionsToGetAccountIDRequest = GetAccountTermsOfServiceRequest()
+        env.api.makeRequest(termsAndConditionsToGetAccountIDRequest) { [weak self] (response, _, error) in
             performUIUpdate {
                 self?.spinner.isHidden = true
                 if let error = error {
                     self?.showError(error)
                 } else {
-                    self?.displayPairingCode(response?.code)
+                    self?.displayQR(pairingCode: pairingCode, accountID: response?.account_id.value, baseURL: self?.env.api.baseURL)
                 }
             }
         }
     }
 
+    func displayQR(pairingCode: String?, accountID: String?, baseURL: URL?) {
+        guard
+            let code = pairingCode,
+            let accountID = accountID,
+            let host = baseURL?.host
+        else { return }
+
+        var comps = URLComponents(string: "canvas-parent://create-account/create-account/\(accountID)/\(code)")
+        comps?.queryItems = [
+            URLQueryItem(name: "baseURL", value: host),
+        ]
+
+        let input = comps?.url?.absoluteString ?? ""
+        let data = input.data(using: String.Encoding.ascii)
+        guard let qrFilter = CIFilter(name: "CIQRCodeGenerator") else { return }
+        qrFilter.setValue(data, forKey: "inputMessage")
```

Comment:
I did not know it was so easy to generate a QR code. I love it.

Reply:
I know, i was pretty surprised it was this easy myself.